### PR TITLE
configuration and flags refactoring

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: cmd/node/config/external.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Explorer]
```
Comment (iulianpascalau) [inline cmd/node/config/external.toml]: Should merge these 2 sections: move the "Explorer" configs inside "ElasticSearchConnector"

File: scripts/testnet/include/config.sh [hunk 2]
```diff
@@ -56,7 +56,6 @@ copyNodeConfig() {
   cp $NODEDIR/config/config.toml ./node/config
   cp $NODEDIR/config/economics.toml ./node/config
   cp $NODEDIR/config/prefs.toml ./node/config
-  cp $NODEDIR/config/server.toml ./node/config
```
Comment (iulianpascalau) [inline scripts/testnet/include/config.sh]: maybe change to copy external.toml

File: cmd/node/config/external.toml (hunk 1 above)
Comment (bogdan-rosianu) [reply cmd/node/config/external.toml]: done

File: scripts/testnet/include/config.sh (hunk 2 above)
Comment (bogdan-rosianu) [reply scripts/testnet/include/config.sh]: done

File: cmd/node/main.go [hunk 3]
```diff
@@ -932,7 +938,7 @@ func copySingleFile(folder string, configFile string) {
 		return
 	}
 	defer func() {
-		err := destination.Close()
+		err = destination.Close()
```
Comment (miiu96) [inline cmd/node/main.go]: go at line 949 and change print message

File: cmd/node/main.go [hunk 4]
```diff
@@ -1034,8 +1040,18 @@ func loadEconomicsConfig(filepath string) (*config.ConfigEconomics, error) {
 	return cfg, nil
 }
 
-func loadPreferencesConfig(filepath string) (*config.ConfigPreferences, error) {
-	cfg := &config.ConfigPreferences{}
+func loadPreferencesConfig(filepath string) (*config.Preferences, error) {
```
Comment (miiu96) [inline cmd/node/main.go]: you should move all that functions from  main.go in a new file maybe a common file or something like that

File: config/tomlConfig_test.go [hunk 5]
```diff
@@ -168,22 +168,54 @@ func TestTomlEconomicsParser(t *testing.T) {
...
-	cfg := ConfigPreferences{}
+	cfg := Preferences{}
 
 	err := toml.Unmarshal([]byte(testString), &cfg)
 
 	assert.Nil(t, err)
 	assert.Equal(t, cfgPreferencesExpected, cfg)
 }
+
+func TestTomlExternalParser(t *testing.T) {
```
Comment (miiu96) [inline config/tomlConfig_test.go]: :+1:

File: cmd/node/main.go (hunk 3 above)
Comment (bogdan-rosianu) [reply cmd/node/main.go]: done

File: cmd/node/main.go (hunk 4 above)
Comment (bogdan-rosianu) [reply cmd/node/main.go]: will leave it like this for now. maybe in a future PR

File: config/tomlConfig_test.go (hunk 5 above)
Comment (bogdan-rosianu) [reply config/tomlConfig_test.go]: :+1:

File: cmd/node/main.go (hunk 4 above)
Comment (miiu96) [reply cmd/node/main.go]: why you cant do this now because you have just to copy a part of the code from main.go in a new file ?

**Pull request merged**